projects/igniteui-angular/src/lib/services/transaction/utilities.ts

```diff
@@ -0,0 +1,104 @@
+export enum TransactionType {
+    ADD = 'add',
+    DELETE = 'delete',
+    UPDATE = 'update'
+}
+
+export interface Transaction {
+    id: any;
+    type: TransactionType;
+    newValue: any;
+}
+
+export interface State {
+    value: any;
+    recordRef: any;
+    type: TransactionType;
+}
+
+export interface TransactionService {
+    /**
+     * Adds provided  transaction with recordRef if any
+     * @param transaction Transaction to be added
+     * @param recordRef Reference to the value of the record in the data source related to the changed item
+     *
+     * @returns If operation is successful.
+     */
+    add(transaction: Transaction, recordRef?: any): boolean;
+
+    /**
+     * Returns an array of all transactions. If id is provided returns last transaction for provided id
+     * @returns All the transaction on last transaction for provided id
+     */
+    getTransactionLog(id?: any): Transaction[] | Transaction;
+
+    /**
+     * Remove the last transaction if any
+     */
+    undo(): void;
+
+    /**
+     * Applies the last undone transaction if any
+     */
+    redo(): void;
+
+    /**
+     * Returns aggregated state of all transactions
+     * @param mergeChanges If set to true will merge each state's value over relate recordRef
+     * and will record resulting value in the related transaction
+     * @returns Collection of aggregated transactions for each changed record
+     */
+    aggregatedState(mergeChanges: boolean): Transaction[];
+
+    /**
+     * Returns the state of the record with provided id
+     * @param id The id of the record
+     * @returns State of the record if any
+     */
+    getState(id: any): State;
+
+    /**
+     * Returns whether transaction is enabled for this service
+     * @returns If transaction is enabled
+     */
+    transactionsEnabled(): boolean;
```

Comment:
@wnvko This should be just `enabled` as a property, where the full implementation would return true always and the  base only when pending collection is in progress.